### PR TITLE
Intermediate tool tutorial

### PR DESCRIPTION
**Pull request by benptc**
This'll probably be the last tool tutorial I write for awhile, but it explains more of the JavaScript APIs in detail and gives advanced users some demos to play around with.

### CODE REVIEW

File: tutorials/context-dependent-tool-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
+## Intermediate Tool Tutorial: Context-dependent Tools
+
+Pre-req: [Creating a Simple Tool](../develop/spatial-tools/tutorial)
+
+Now that you know how to create a tool and interact with some basic RealityInterface APIs,
+we're going to explore some more interesting things you can do with tools. This tutorial gets a
+bit complicated, but as always please [use the forum](forum.spatialtoolbox.vuforia.com) if you
```
Comment (hobinjk-ptc) [inline tutorials/context-dependent-tool-tutorial.md]: This needs an https:// in front of it

File: tutorials/context-dependent-tool-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+its 15th entry (`modelView[14]`).
+
+##### Demo 1: Adjust a continuous property based on distance
+
+In this first example, we continuously adjust the color of the tool as you move from about 0.5
+meters away from the tool to about 1 meter away from the tool:
+
+```javascript
+realityInterface.addMatrixListener(function(modelView, _projection) {
+  	// convert the scary matrix into easy-to-use values
```
Comment (hobinjk-ptc) [inline tutorials/context-dependent-tool-tutorial.md]: mixed tabs and spaces are making this look a bit weird

Comment (benptc) [reply tutorials/context-dependent-tool-tutorial.md]: good catch! converted them all to spaces.

File: tutorials/context-dependent-tool-tutorial.md [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 Pre-req: [Creating a Simple Tool](../develop/spatial-tools/tutorial)
 
 Now that you know how to create a tool and interact with some basic RealityInterface APIs,
 we're going to explore some more interesting things you can do with tools. This tutorial gets a
-bit complicated, but as always please [use the forum](forum.spatialtoolbox.vuforia.com) if you
-have any questions.
+bit complicated, but as always please [use the forum](https://forum.spatialtoolbox.vuforia.com) if
+you have any questions.
 
 ##### Setting Up
```
Comment (hobinjk-ptc) [inline tutorials/context-dependent-tool-tutorial.md]: I think this might need to be ### instead. Sorry for not catching this earlier!

Comment (benptc) [reply tutorials/context-dependent-tool-tutorial.md]: oops I merged too soon. I can fix in a follow-up

**Pull request merged**